### PR TITLE
fix: Fixed header layout on mobile devices

### LINKED ISSUE #296: Improve header layout on mobile
## Use Case

Colorizer looks bad in mobile browsers. Low hanging fruit would be to fix the header, which is currently way to wide on smaller screens.

## Acceptance Criteria

_Please describe how you know this is done_

## Details

_Please provide any helpful specifications_

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
Closes #296, "improving header layout on mobile", as well as sprinkling in a few other fixes.

*Estimated review size: small, 15-20 minutes*

Solution
========
- Simplifies `FlexDiv` utility class and adds a prop for wrapping behavior.
- Fixes layout issue on the viewer page where the Canvas would not match screen width.
  -  EXPERIMENTALLY allows for resizing the canvas height too to maintain aspect ratio. See TODOs for possible issues with this approach.
- Header changes
  - Adds a smaller version of the AICS logo, which will be used on mobile devices.
  - Allowed elements in the header to wrap, rather than forcing the header to extend past the screen bounds.

## Type of change
* Bug fix (non-breaking change which fixes an issue)
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-297/
2. Navigate to the viewer.
3. (On Chrome) press F12 to open the developer console. Open the device toolbar.
image

4. Click through a couple different device sizes to confirm the header and main canvas area look reasonable across them.

Screenshots (optional):
-----------------------
**Video talkthrough (🔊):**

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/fa0c1a80-b5f6-4148-8b1b-d3f7f96fec6c

image

### CODE REVIEW

File: src/Viewer.module.css [hunk 1]
```diff
@@ -12,11 +12,10 @@
   flex-direction: row;
 }
 
-@media screen and (max-width: 1200px) {
+@media screen and (max-width: 1250px) {
```
Comment (ShrimpCryptid) [inline src/Viewer.module.css:15]: See note about this change in `CanvasWrapper.tsx`.

File: src/Viewer.module.css [hunk 2]
```diff
@@ -51,8 +51,7 @@
 }
 
 .canvasPanel {
-  flex-grow: 0;
-  max-width: var(--content-width);
+  width: var(--content-width);
```
Comment (ShrimpCryptid) [inline src/Viewer.module.css:54]: Prevents compression from other components fighting with the canvas panel for space.

File: src/components/CanvasWrapper.tsx [hunk 3]
```diff
@@ -236,10 +237,21 @@ export default function CanvasWrapper(inputProps: CanvasWrapperProps): ReactElem
      * TODO: Margin calculation?
      */
     const setSize = (): void => {
+      // TODO: Potentially unsafe calculation here when using `window.innerWidth`. If close to the breakpoint where the side
+      // panel gets wrapped to below the canvas, the scrollbar added to account for the increased page height
+      // will cause this calculation to change (window.innerWidth will become smaller by ~15 pixels).
+      // Under certain circumstances, this can cause a flickering effect as the canvas resizes to accommodate the scrollbar,
+      // which causes the page to shrink, which causes the scrollbar to disappear, and so on in a loop.
+      // I've fixed this for now by setting the breakpoint to 1250 pixels, but it's not a robust solution.
+
```
Comment (ShrimpCryptid) [inline src/components/CanvasWrapper.tsx:246]: Example of what this looks like:

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/50cdbb9b-1628-4800-8929-0bf17724c738

**Review by frasercl — APPROVED** [review]
Looks good!

**Review by toloudis — APPROVED** [review]
LGTM!

**Pull request merged**